### ISSUE TITLE
An experiment about the benefits of ELF

### ISSUE THREAD

**Issue by bjiyxo** [gcp/leela-zero#1499]
I want to confirm the benefits of ELF, so I trained a network from ecab83bb (131), which hadn't trained any ELF self-play, and trained it with the training data 121-130 & ELF selfplay (5/24 version, so it is ~75k games), so the window is ~500k. The batch size is 512, the learning rate is 0.0001, and run 64k steps. I called it ecab-t (t stands for two, haha my English is really poor.) Here is the weight.
https://drive.google.com/file/d/1TwyHP0coYNALQGE278bjFUPWMrXy7ZJV/view?usp=sharing

And I tested ecab83bb (131) with ecab-t. Here are the results,
[ecabvsecab-t.zip](https://github.com/gcp/leela-zero/files/2060093/ecabvsecab-t.zip)
**ecab83bb (131) : ecab-t = 91 : 109**, so the one with ELF has 54.5% win rate. Meanwhile, we have progressed over 10 times, but elf's selfplay may not guarantee us to win 55% once.

Now we can confirm that the benefit from ELF is little comparing to the noise correction, and the recent dramatic growing is because of the noise correction, not ELF.

**Comment by alphaladder** [comment]
Need the third one without elf  but only 121-130 as control

**Comment by bjiyxo** [comment]
It will be like ecab(131), and the result won't change dramatically in my opinion.

**Comment by l1t1** [comment]
please copy it to another cloud driver, thanks @bjiyxo

**Comment by bjiyxo** [comment]
@l1t1 Contact me on QQ 2868317510.

**Comment by l1t1** [comment]
I test the 73a1 weight [http://storage.sjeng.org/networks/73a12e721c668db435794b38304e686550974d2cb2895280bfdc69699afc10d5.gz], which is siblings of ecab and almost as strong as ecab (54.34 winrate to their parent), and has no elf blood


2018-05-04 21:23 | d987c1c9 7.176M+192k192x15 VS Regular Match ecab83bb 7.176M+96.0k192x15 | 188 : 176 (51.65%) | 364 / 400 | fail
-- | -- | -- | -- | --
2018-05-04 12:18 | 73a12e72 7.176M+128k192x15 VS Regular Match 18e6a6c5 7.100M+256k192x15 | 219 : 184 (54.34%) | 403 / 400 | fail
2018-05-04 07:45 | ecab83bb 7.176M+96.0k192x15 VS Regular Match 18e6a6c5 7.100M+256k192x15 | 226 : 186 (54.85%) | 412 / 400 | fail

**Comment by gcp** [comment]
I'll do a training run with the ELF data removed this weekend to see what happens.

**Comment by l1t1** [comment]
```
# Black: Leela Zero
# BlackCommand: D:\tool\valid14\leelaz.exe -g -w D:\ecab-t.gz -v3200 --noponder -m4 -t1 -s1
# BlackLabel: Leela Zero:0.15
# BlackVersion: 0.15
# Date: June 1, 2018 1:06:23 PM CST
# Host: PC
# Komi: 7.5
# Referee: -
# Size: 19
# White: Leela Zero
# WhiteCommand: D:\tool\valid14\leelaz.exe -g -w D:\tool\valid14\73a1.gz  -v3200 --noponder -m4 -t1 -s1
# WhiteLabel: Leela Zero:0.15
# WhiteVersion: 0.15
# Xml: 0
#
#GAME	RES_B	RES_W	RES_R	ALT	DUP	LEN	TIME_B	TIME_W	CPU_B	CPU_W	ERR	ERR_MSG
0	W+R	W+R	W+R	0	-	230	2690.8	2656.1	0	0	0	
```
[131t_73a1-0.sgf.txt](https://github.com/gcp/leela-zero/files/2061488/131t_73a1-0.sgf.txt)

**Comment by l1t1** [comment]
@gcp  Did run with the ELF data removed mean remove all data after LZ131 ?

**Comment by gcp** [comment]
No just remove the ELF data from the training window. The later networks were strong so of course I'll include their data.

**Comment by Tsun-hung** [comment]
If I remenber correctly, the promotion slow down at network of 905 period, after noise bug is fixed, it increases quickly again. So I think the elf data is beneficial between eca and 905, the second big jump latest is attributed to bugfix.

**Comment by PhilipFRipper** [comment]
it's learning a wider variety of positions, so it has a lot of lateral learning going on, which won't translate well to winrate, but will translate to winrate against more varied opponents like humans.  At least, in theory.

If it cannot learn from observing a network that is much stronger than it, it would kind of disprove the whole thing we've already proven: that this works.  So that makes no sense to me.  I think if we are learning from it, but not improving, then we must have been in local minima that the training is having to dig us out from before we can improve.  That's my hypothesis, but the only way I know to test it is to take another month or two doing what we're already doing.

**Comment by l1t1** [comment]
as my result, ecab-t is weaker than 73a1

**Comment by bjiyxo** [comment]
@l1t1 Are you going to judge the strength between ecab-t & 73a1 with only 4 games?

**Comment by l1t1** [comment]
from above test, it seems  the strength of  ecab-t & 73a1 & lz147 are similar

**Comment by l1t1** [comment]
@gcp did the training run with the ELF data removed start?

**Comment by PhilipFRipper** [comment]
@l1t1 With a confidence interval of 95%, a sample size of ten has a margin of error of more than 30% of the sample.  So your 6 wins means between 3 and 9, and 4 wins is between 1 and 7 -- with no way to tell where in that it is.  It's within the 95% probability that your 6:4 represents 3:7.  Anyway, never use sample sizes that small they don't do anything.    A sample size of 100 has a 95% confidence interval MoE of +/- 10.  If you want something more useful to work with.

**Comment by l1t1** [comment]
@PhilipFRipper @bjiyxo you are right, i will just do autogtp in the future.

**Comment by gcp** [comment]
The networks that are going up now are with the ELF data removed. As you can see they seem to be worse, so far.

**Comment by bjiyxo** [comment]
@gcp Do you mean e3654ad6 is the network with the ELF data removed?

**Comment by l1t1** [comment]
does http://zero.sjeng.org/networks/5b59a01883f45abed1cb336bffd89cbf6618398f84930bf03d096252ce8b41a1.gz remove elf data? it's the strongest  lz weight as so far

**Comment by gcp** [comment]
2f2e2b82 was the first one without, this will continue until 256k steps as usual. But the grouping of dots seems to be a bit below the previous cluster, so ELF does seem to help a bit.

**Comment by gcp** [comment]
>it  sounds like if ELF games are helping so much then maybe we should start  trying to ween LZ off of elf games by slowly reducing the percentage  each passed network.

As already said, the plan is to continue until about 50% of the window, at which point LZ is hopefully rather close to ELF strength.

**Comment by l1t1** [comment]
hard to understand.
as you say, 2f2e and latter only use 1z147 selfplay games, do not use elf games, since elf games is useless for trainning. why continue to send tasks to client? only keep those games for study? or once find only use 147 games cannot training weight better than 147 in a very long peroid, feed it elf games again?

**Comment by PhilipFRipper** [comment]
@l1t1 

People wanted to test if ELF helped.  So GCP ran a cycle w/o ELF games.  That cycle is weaker.  So it shows ELF games are useful.  So we are going back to what we were doing before.  ELF games are useful for training and do help.  This was just an experiment of one cycle.

**Comment by WYVYW** [comment]
Would it then make sense to feed the last 128x10 with the self-play games from the 192x15 net to see how much games from a stronger net help and to get a stronger 128x10 net for mobile devices?

**Comment by l1t1** [comment]
@WYVYW you can training it and compare it to one 129x15 weight

**Comment by roy7** [comment]
@gcp Was d018 with ELF back in there, or without ELF data?

**Comment by FeoDoT90** [comment]
I think the size of the nets is not important, but the quality of the food provided for the net is important. For every million game samples, it will take another two million games to train, so that the network can significantly Refine its coefficients. In my opinion, this is logical, because here the key role is played by science - statistics. Therefore it is impossible to adopt the accumulated experience of old networks, for networks of new and large sizes. Otherwise, we rely only on luck that one day the network will appreciate the once rejected course of action from numerous weak experiences. After all, the essence of training is not to feed the network a lot of games generated exclusively by accident, or human experience, or a competitor's game, but to give it a little more powerful experience that the network can correctly learn.  If we continue to invest in the existing experience of existing networks, we will grow a very short-sighted network, and I think we will have to make much more effort than it takes to reach level of AGZ. Just rate the growth graph of the force LZ, he is not like the progress chart of AGZ. 

My moods can be very depressing and radical, but I urge you to give up the old network experience, and start from scratch to reach higher heights in a few months, with the final size of the 40x256 network.

I think we need to forget about the artificial ELF network, the origin of which we know nothing about, and stop chasing other competitors. We have already done a miracle worthy of admiration, but I think it is necessary to settle down before it's too late, to abandon the high-speed race, and start our flight of strength growth with absolute chaos of zero experience.
